src/goto-programs/graphml_witness.cpp

```diff
@@ -47,13 +61,63 @@ void graphml_witnesst::remove_l0_l1(exprt &expr)
     remove_l0_l1(*it);
 }
 
+template <typename T>
+inline void hash_combine(std::size_t &seed, const T &v)
+{
+  std::hash<T> hasher;
+  seed ^= hasher(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+}
+
+namespace std // NOLINT(readability/namespace)
+{
+template <typename S, typename T>
+struct hash<pair<S, T>> // NOLINT(readability/identifiers)
+{
+  inline size_t operator()(const pair<S, T> &v) const
+  {
+    size_t seed = 0;
+    ::hash_combine(seed, v.first);
+    ::hash_combine(seed, v.second);
+    return seed;
+  }
+};
+} // namespace std
+
 std::string graphml_witnesst::convert_assign_rec(
   const irep_idt &identifier,
   const code_assignt &assign)
 {
+  static std::
```

Comment:
Quite surprised the lifetime of this cache isn't tied to that of some class, rather than living as long as the process

Reply:
I'll make it a class member.